nixos/modules/virtualisation/lxc.nix

```diff
@@ -73,5 +93,26 @@ in
         include ${cfg.package}/etc/apparmor.d/lxc-containers
       '';
     };
+
+    # `lxc-user-nic` needs suid to attach to bridge for unpriv containers.
+    security.wrappers = lib.mkIf cfg.unpriv {
+      lxcUserNet = {
+        source = "${pkgs.lxc}/libexec/lxc/lxc-user-nic";
+        setuid = true;
+        owner = "root";
+        group = "root";
+        program = "lxc-user-nic";
+      };
+    };
+
+    # Add lxc-net service if unpriv mode is enabled.
+    systemd.packages = lib.mkIf cfg.unpriv [ pkgs.lxc ];
+    systemd.services = lib.mkIf cfg.unpriv {
```

Comment:
This creates a service with no Exec. Is that on purpose?

Reply:
lxc has a [service file](https://github.com/lxc/lxc/blob/main/config/init/systemd/lxc-net.service.in), which is imported on line 109, but i noticed that it won't start on boot. To solve that i added this service to "override" some config and enable it.